example/sparse_advection/sparse_advection_driver.cpp

```diff
@@ -122,12 +122,12 @@ TaskCollection SparseAdvectionDriver::MakeTaskCollection(BlockList_t &blocks,
                              mdudt.get(), beta * dt, mc1.get());
 
     // do boundary exchange
-    auto restrict =
+    auto prolong_restrict =
```

Comment:
`restrict` is a C keyword, and was not compiling for me. I'm actually not sure how we've been getting away with this for months...

Reply:
huh. Good catch.

Reply:
I think it worked because `restrict` is only sort of a keyword, see https://github.com/parthenon-hpc-lab/parthenon/pull/1009#issuecomment-2009801991.

Reply:
I agree it should be changed, but the change may conflict with #1009 where it is already changed. Simple enough to deal with though.

Reply:
Merge conflict resolved :+1: